src/lightning_app/core/queues.py

```diff
@@ -291,3 +301,98 @@ def is_running(self) -> bool:
             return self.redis.ping()
         except redis.exceptions.ConnectionError:
             return False
+
+
+class HTTPQueue(BaseQueue):
+    def __init__(self, name: str, default_timeout: float):
+        """
+        Parameters
+        ----------
+        name:
+            The name of the Queue to use. In the current implementation, we expect the name to be of the format
+            `appID_queueName`. Based on this assumption, we try to fetch the app id and the queue name by splitting
+            the `name` argument.
+        default_timeout:
+            Default timeout for redis read
+        """
+        if name is None:
+            raise ValueError("You must specify a name for the queue")
+        self.app_id, self.name = self._split_app_id_and_queue_name(name)
+        self._original_name = name  # keeping the name for debugging
+        self.default_timeout = default_timeout
+        self.client = HTTPClient(base_url=HTTP_QUEUE_URL, log_callback=debug_log_callback)
+
+    def get(self, timeout: int = None) -> Any:
+        if not self.app_id:
+            raise ValueError(f"App ID couldn't be extracted from the queue name: {self._original_name}")
+
+        # it's a blocking call, we need to loop and call the backend to mimic this behavior
+        if timeout is None:
+            while True:
+                try:
+                    return self._get()
+                except queue.Empty:
+                    time.sleep(HTTP_QUEUE_REFRESH_INTERVAL)
+
+        # make one request and return the result
+        if timeout == 0:
+            return self._get()
+
+        # timeout is some value - loop until the timeout is reached
+        start_time = time.time()
+        timeout += 0.1  # add 0.1 seconds as a safe margin
+        while (time.time() - start_time) < timeout:
+            try:
+                return self._get()
+            except queue.Empty:
+                time.sleep(HTTP_QUEUE_REFRESH_INTERVAL)
+
+    def _get(self):
+        resp = self.client.post(f"v1/{self.app_id}/{self.name}", query_params={"action": "pop"})
+        if resp.status_code == 204:
+            raise queue.Empty
+        return pickle.loads(resp.content)
+
+    def put(self, item: Any) -> None:
+        if not self.app_id:
+            raise ValueError(f"App ID couldn't be extracted from the queue name: {self._original_name}")
+
+        value = pickle.dumps(item)
+        queue_len = self.length()
+        if queue_len >= WARNING_QUEUE_SIZE:
+            warnings.warn(
+                f"The Queue {self.name} length is larger than the recommended length of {WARNING_QUEUE_SIZE}. "
+                f"Found {queue_len}. This might cause your application to crash, please investigate this."
+            )
+        resp = self.client.post(f"v1/{self.app_id}/{self.name}", data=value, query_params={"action": "push"})
```

Comment:
Should this include projects?

Reply:
We don't really need to include the project ID technically, as far as app ID is unique across the system and this isn't really following REST, so no real need to arrange the resources hierarchically.